Sources/OpenTelemetry/Tracing/Processing/OTelNoOpSpanProcessor.swift

```diff
@@ -0,0 +1,36 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2023 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import ServiceContextModule
+
+/// A span processor that ignores all operations, used when no spans should be processed.
+public struct OTelNoOpSpanProcessor: OTelSpanProcessor {
+    /// Initialize a no-op span processor.
+    public init() {}
+
+    public func run() async throws {
+        while !Task.isCancelled {}
```

Comment:
This is an infinite loop without sleep

Reply:
Good point, I'm not really sure how to best implement no-op services like this with `ServiceLifecycle` 🤔

This is what I came up with, but I'm not very happy about it. It seems like a waste of resources, but at least it's cleanly shut down whenever the service group is being shut down.

```swift
public struct OTelNoOpSpanProcessor: OTelSpanProcessor {
    private let stream: AsyncStream<Void>
    private let continuation: AsyncStream<Void>.Continuation

    /// Initialize a no-op span processor.
    public init() {
        (stream, continuation) = AsyncStream.makeStream()
    }

    public func run() async throws {
        continuation.yield()
        var sequence = stream.cancelOnGracefulShutdown().makeAsyncIterator()
        while await sequence.next() != nil {
            continuation.yield()
        }
    }
}
```

@FranzBusch Do you have an idea on how to implement these no-op services?

Reply:
Your proposed code is fine. Not the end of the world to create an `AsyncStream` here. If you wanna be fancy you can solve it with a continuation.

Reply:
You can do the same with just a single `withCheckedContinuation` rather than the stream. But yeah, this kind of dance is preferred generally

Reply:
Thanks for the input 🙏 I went with the `AsyncStream`-approach for now.